src/modules/0.0.23/aws_vpc/mappers/route_table.ts

```diff
@@ -0,0 +1,258 @@
+    create: async (es: RouteTable[], ctx: Context) => {
+      const out: RouteTable[] = [];
+      for (const e of es) {
+        const client = (await ctx.getAwsClient(e.region)) as AWS;
+
+        // create route table
+        const rawRouteTable = await this.createRouteTable(client.ec2client, e.vpc.vpcId!, e.tags);
+        if (!rawRouteTable) continue;
+        const routeTable = await this.routeTableMapper(rawRouteTable, e.region, ctx);
+        if (!routeTable) continue;
+
+        // create routes
+        e.routes?.map(async r => {
+          if (r.GatewayId === 'local') return; // created by AWS, can't be created by the user
+          await RouteTableMapper.createRoute(client.ec2client, routeTable, r);
+        });
+
+        routeTable.id = e.id;
+        await this.module.routeTable.db.update(routeTable, ctx);
+        out.push(routeTable);
+      }
+      return out;
+    },
+    read: async (ctx: Context, id?: string) => {
+      const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
+      if (!!id) {
+        const { routeTableId, region } = this.idFields(id);
+        const client = (await ctx.getAwsClient(region)) as AWS;
+        const rawRouteTable = await this.getRouteTable(client.ec2client, routeTableId);
+        if (!rawRouteTable) return;
+        return await this.routeTableMapper(rawRouteTable, region, ctx);
+      } else {
+        const out: RouteTable[] = [];
+        await Promise.all(
+          enabledRegions.map(async region => {
+            const client = (await ctx.getAwsClient(region)) as AWS;
+            for (const rawRouteTable of await this.getRouteTables(client.ec2client)) {
+              const routeTable = await this.routeTableMapper(rawRouteTable, region, ctx);
+              if (!routeTable) continue;
+              out.push(routeTable);
+            }
+          }),
+        );
+        return out;
+      }
+    },
+    update: async (es: RouteTable[], ctx: Context) => {
+      const out = [];
+      for (const e of es) {
+        const cloudRecord: RouteTable = ctx?.memo?.cloud?.RouteTable?.[this.entityId(e)];
+
+        if (cloudRecord.vpc.vpcId !== e.vpc.vpcId) {
+          // delete and create a new one
+          await this.module.routeTable.cloud.delete(cloudRecord, ctx);
+          out.push((await this.module.routeTable.cloud.create(e, ctx)) as RouteTable);
+          continue;
+        }
+
+        const client = (await ctx.getAwsClient(e.region)) as AWS;
+        if (!this.eqListItems<Route>(this.eqRoute, e.routes, cloudRecord.routes)) {
+          // delete and create routes
+          const { entitiesInDbOnly, entitiesInAwsOnly } = this.findRoutesDiff(e.routes, cloudRecord.routes);
+          await Promise.all([
+            ...entitiesInAwsOnly.map(async r => await RouteTableMapper.deleteRoute(client.ec2client, e, r)),
+            ...entitiesInDbOnly.map(async r => await RouteTableMapper.createRoute(client.ec2client, e, r)),
+          ]);
+        }
```

Comment:
You may want to do this with two `for` loops instead of a `Promise.all`. The `Promise.all` will trigger the promises near-simultaneously and will be more vulnerable to rate limiting checks for larger tables.